torchrl/data/replay_buffers/storages.py

```diff
@@ -149,7 +149,7 @@ def _rand_given_ndim(self, batch_size):
         if self.ndim == 1:
             return torch.randint(
                 0,
-                len(self),
+                self._len,
```

Comment:
For some reason, the `len(self)` call here was causing recompiles. Changing it to `self._len` avoids the recompiles, even though `TensorStorage.__len__` just returns `self._len` anyway

Reply:
Though I can see that this change made other tests fail, so this isn't a good solution